### PR TITLE
HiPay: Scrub/Refund/Void

### PR DESCRIPTION
**Pull request by gasb150**
SER-720
SER-721
----
Summary:
----
Adding to the HiPay gateway support for scrub, refund, and void.

HiPay allows partial capture and partial refunds, this commit include the amount to use it.

Tests
----
Remote Test:
Finished in 6.627757 seconds.
15 tests, 62 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

Unit Tests:
21 tests, 61 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications

RuboCop:
787 files inspected, no offenses detected

### CODE REVIEW

File: lib/active_merchant/billing/gateways/hi_pay.rb [hunk 1]
```diff
@@ -51,23 +51,42 @@ def authorize(money, payment_method, options = {})
...
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )[\w =]+), '\1[FILTERED]').
+          gsub(%r((card_number=)\w+), '\1[FILTERED]\2').
+          gsub(%r((cvc=)\w+), '\1[FILTERED]\2')
       end
 
       private
 
+      def maintenance_operation(money, authorization, options)
```
Comment (aenand) [inline lib/active_merchant/billing/gateways/hi_pay.rb:82]: Can we rename this to be something a bit more specific. Maybe `reference_operation`?

**Review by aenand — COMMENTED** [review]
This looks good to me! Just one request for re-naming

File: lib/active_merchant/billing/gateways/hi_pay.rb (hunk 1 above)
Comment (gasb150) [reply lib/active_merchant/billing/gateways/hi_pay.rb:82]: In this case, the name was according to those operations' names in the documentation. https://developer.hipay.com/api-explorer/api-online-payments#/maintenance/performMaintenanceOperation.

But yes, I can rename it.

**Review by aenand — APPROVED** [review]
LGTM :shipit:

**Review by naashton — APPROVED** [review]

**Pull request merged**